force-app/main/default/classes/CourseRegistrationHandler.cls

```diff
@@ -0,0 +1,219 @@
+        List<String> coursesReduceWaiting = new List<String>();
+
+        for (CourseRegistration__c cr : registrations) {
+            Boolean reducedAttending = cr.Status__c == 'Påmeldt';
+            Boolean reducedWaiting = cr.Status__c == 'Venteliste';
+
+            if(reducedAttending){
+                coursesReduceAttending.add(cr.Course__c);
+            }
+            else if(reducedWaiting){
+                coursesReduceWaiting.add(cr.Course__c);
+            }
+        }
+        if(!coursesReduceAttending.isEmpty()){
+            countCourseIdsToReduceAttending(coursesReduceAttending);
+        }    
+        if(!coursesReduceWaiting.isEmpty()){
+            countCourseIdsToReduceWaiting(coursesReduceWaiting);
+        }
+    }
+
+    public static void getCourseIdToIncrease(List<CourseRegistration__c> registrations) {
+
+        List<String> coursesIncreaseAttending = new List<String>();
+        List<String> coursesIncreaseWaiting = new List<String>();
+
+        for (CourseRegistration__c cr : registrations) {
+            Boolean increasedAttending = cr.Status__c == 'Påmeldt';
+            Boolean increasedWaiting = cr.Status__c == 'Venteliste';
+            if(increasedAttending){
+                coursesIncreaseAttending.add(cr.Course__c);
+            }
+            else if(increasedWaiting){
+                coursesIncreaseWaiting.add(cr.Course__c);
+            }
+        }   
+        if(!coursesIncreaseAttending.isEmpty()){
+            countCourseIdsToIncreaseAttending(coursesIncreaseAttending);
+        }
+        if(!coursesIncreaseWaiting.isEmpty()){
+            countCourseIdsToIncreaseWaiting(coursesIncreaseWaiting);
+        }
+    }
+
+    public static void updateCoursesParticipants(Map<String,Integer> coursesCount){
+        List<Course__c> coursesToUpdate = new List<Course__c>();
+        for(Course__c course : [SELECT Id, NumberOfParticipants__c FROM Course__c WHERE Id IN :coursesCount.keyset()]) {
+            course.NumberOfParticipants__c = course.NumberOfParticipants__c + coursesCount.get(course.Id);
+            coursesToUpdate.add(course);
+        }
+        update coursesToUpdate;
+    }
+
+    public static void updateCourseWaiting(Map<String,Integer> coursesCountWait){
+        List<Course__c> coursesToUpdateWait = new List<Course__c>();
+        for(Course__c course : [SELECT Id, Waitinglist__c FROM Course__c WHERE Id IN :coursesCountWait.keyset()]) {
+            course.Waitinglist__c = course.Waitinglist__c + coursesCountWait.get(course.Id);
+            coursesToUpdateWait.add(course);
+        }
+        update coursesToUpdateWait;
+    }
+
+    public static void countCourseIdsToReduceAttending(List<String> reduceAttending) {
+        Map<String,Integer> reduceCountIds = new Map<String,Integer>();
+        for(String id : reduceAttending) {
+            if(!reduceCountIds.containsKey(id)) {
+                reduceCountIds.put(id,0);
+            }
```

Comment:
Hva skjer her hvis du har to CourseParticipants som skal anmeldes samtidig, på samme kurs? Kurset vil bare reduseres med én og ikke to stk. Se testene lengre nede 

**SKRIV TESTENE FØRST, OG SE AT DE BLIR RØDE FØR DU FIKSER FEILEN**